Fix Vert.x and Hibernate validator examples

antoinesd opened the pull request:
fixes #206 
JLink image (in JLink profile) is now working. Produced image is also functioning.

aalmiray commented on integrationtest/vert.x/pom.xml:
```diff
@@ -286,16 +270,6 @@
                                     </artifact>
                                     <moduleInfo>
                                         <name>io.vertx.core</name>
-                                        <requires>
```
Why were these customizations removed?

antoinesd replied:
Because jdeps adds them automatically so it makes a lighter config

aalmiray commented on integrationtest/vert.x/pom.xml:
```diff
@@ -307,12 +281,10 @@
-                                        com.example to io.vertx.core;
-                                    </exports>
-                                </moduleInfo>
+                                <moduleInfoSource>open module com.example {
```
Does the module have to be open?

antoinesd replied:
Same answer here.
The main change is in the `<jdepsExtraArgs>` that allows jdeps to be less dumb by analyzing modules already created by the plugin. It's the reason why I change some module order : to have dependent module computed by jdeps after upstream ones

antoinesd replied:
Have you tested the jlink image @aalmiray ? I'm not saying that my config is totally ok but it produces a working result. I'm not expert enough in Vert.x to go further : we should have a more complex app example to fine tune the config automatically computed by `jdeps`

aalmiray replied:
Granted, the use of `open` makes everything in the module visible to outsiders. The error you see could also be fixed by explicitly exporting the required package to the vertx module. 

Open modules should not be used by libraries but I suppose it's the easy way for applications ad it's this case.

antoinesd replied:
So I should change it to an export mention I guess. Will try that

antoinesd commented:
BTW, the docker part of the example is still not working at the moment.
I also tried to update Vert.x and Netty version but it brought a lot of other corner case.
I'll try to fix these later if the PR is merged

aalmiray commented:
@antoinesd could you rebase from your side? I thought I could fix a merge conflict from mine side but something went wrong, sorry 😓

antoinesd commented:
I force-pushed, changed the title and I'm working on resolving build failure for hibernate validator example

antoinesd commented:
@aalmiray, @gunnarmorling : Now the build pass.
- I corrected all license mention even in the files not covered by the licence plugin, 
- I removed parent pom path and packaging from examples
- and last but not least (the most brain f*** part) I updated Hibernate Validator example to Jakarta EE 10 and make it pass the build (didn't try to launch the resulting artifact : not sure what is expected with this test)

aalmiray commented on .github/workflows/build.yml:
```diff
@@ -1,18 +1,8 @@
+#  PDX-License-Identifier: Apache-2.0
```
Missing a leading `S` in all modified headers

antoinesd replied:
Sorry for that

aalmiray commented on integrationtest/hibernate-validator/pom.xml:
```diff
@@ -7,172 +7,211 @@
+                  </artifact>
+                  <moduleInfoSource>
+                    module paranamer {
+                    requires static javax.inject;
```
Please indent this line with 4 spaces to the right 😅

aalmiray commented on integrationtest/hibernate-validator/pom.xml:
```diff
@@ -7,172 +7,211 @@
+              <module>
+                <mainClass>com.example.ValidationTest</mainClass>
+                <moduleInfoSource>open module com.example {
+                  requires transitive org.hibernate.validator;
```
Please indent this line also.

aalmiray commented on integrationtest/vert.x/pom.xml:
```diff
@@ -299,12 +271,11 @@
-                                </moduleInfo>
+                                <moduleInfoSource>module com.example {
+                                    exports com.example to io.vertx.core;
+                                    requires io.vertx.core;
```
Please indent this line

antoinesd replied:
I changed all my ident to 4 chars

aalmiray commented:
LGTM @gunnarmorling do you have any additional comments?
